lib/internal/Magento/Framework/Setup/Declaration/Schema/SchemaConfig.php

```diff
@@ -72,10 +72,18 @@ public function getDbConfig()
      */
     public function getDeclarationConfig()
     {
-        $schema = $this->schemaFactory->create();
-        $data = $this->readerComposite->read(FileResolverByModule::ALL_MODULES);
-        $this->declarativeSchemaBuilder->addTablesData($data['table']);
-        $schema = $this->declarativeSchemaBuilder->build($schema);
-        return $schema;
+        try{
+            $schema = $this->schemaFactory->create();
+            $data = $this->readerComposite->read(FileResolverByModule::ALL_MODULES);
+            $this->declarativeSchemaBuilder->addTablesData($data['table']);
+            $schema = $this->declarativeSchemaBuilder->build($schema);
+            return $schema;
+        }catch (\Exception $e) {
+            $writer = new \Zend\Log\Writer\Stream(BP . '/var/log/schema.log');
+            $logger = new \Zend\Log\Logger();
+            $logger->addWriter($writer);
+            $logger->info($e->getMessage());
+            throw new \Exception($e->getMessage());
```

Comment:
What it it's a child class that is caught? something that was more specific. You just changed the class and lost the stack trace.